test/widgets/search_ride_page_test.dart

```diff
@@ -0,0 +1,967 @@
+          expect(pageState.destinationController.text, formerStart);
+        });
+      });
+
+      group('Enter date', () {
+        group('Whole day', () {
+          testWidgets('Via arrows', (WidgetTester tester) async {
+            final int dayDifference = faker.randomGenerator.integer(4);
+            final DateTime dateTime = DateTime.now().add(Duration(days: dayDifference));
+
+            await pumpMaterial(tester, const SearchRidePage());
+            await tester.pump();
+
+            final SearchRidePageState pageState = tester.state(pageFinder);
+
+            for (int i = 0; i < dayDifference + 1; i++) {
+              await tester.tap(find.byKey(const Key('searchRideAfterButton')));
+              await tester.pump();
+            }
+            await tester.tap(find.byKey(const Key('searchRideBeforeButton')));
+            await tester.pump();
+
+            expect(pageState.selectedDate.year, dateTime.year);
+            expect(pageState.selectedDate.month, dateTime.month);
+            expect(pageState.selectedDate.day, dateTime.day);
+          });
+
+          testWidgets('Via DatePicker', (WidgetTester tester) async {
+            final int dayDifference = faker.randomGenerator.integer(4);
+            final DateTime dateTime = DateTime.now().add(Duration(days: dayDifference));
+
+            await pumpMaterial(tester, const SearchRidePage());
+            await tester.pump();
+
+            final SearchRidePageState pageState = tester.state(pageFinder);
+
+            await enterDate(tester, dateTime);
+
+            expect(pageState.selectedDate.year, dateTime.year);
+            expect(pageState.selectedDate.month, dateTime.month);
+            expect(pageState.selectedDate.day, dateTime.day);
+          });
+        });
+
+        group('Not whole day', () {
+          testWidgets('Possible time', (WidgetTester tester) async {
+            final DateTime dateTime = DateTime.now().add(Duration(
+              days: random.integer(4),
+              hours: random.integer(23),
+              minutes: random.integer(59),
+            ));
+
+            await pumpMaterial(tester, const SearchRidePage());
+            await tester.pump();
+
+            final SearchRidePageState pageState = tester.state(pageFinder);
+
+            await enterDateAndTime(tester, dateTime);
+
+            expect(pageState.selectedDate.year, dateTime.year);
+            expect(pageState.selectedDate.month, dateTime.month);
+            expect(pageState.selectedDate.day, dateTime.day);
+            expect(pageState.selectedDate.hour, dateTime.hour);
+            expect(pageState.selectedDate.minute, dateTime.minute);
+          });
+
+          testWidgets('Impossible time', (WidgetTester tester) async {
+            final DateTime dateTime = DateTime.now().subtract(const Duration(minutes: 11));
+            if (dateTime.day != DateTime.now().day) {
+              // Code is untestable from 00:00 to 00:11
```

Comment:
I know we talked about this... But I am unhappy that the line is just not part of the coverage when we run it at that time - I feel like coverage should be predetermined

Reply:
Yes I'm not happy with it either, but I didn't want to implement anything before we talked about it because I think the solution isn't beautiful and it's important for further development. As discussed [here](https://stackoverflow.com/questions/53816067/how-to-test-code-that-uses-datetime-now-in-flutter), we could
1. use the dart `clock` class instead of `DateTime`. I don't like that because I feel it's not very Flutter-y, everywhere I look everyone uses `DateTime`.
2. implement an extension method on `DateTime` and use that. I don't like that because I don't think there's a good way to make the difference between a mockable now and an un-mockable now very clear, and people will be wondering why we used a strange method at first sight. Besides, we'd essentially be re-implementing `clock`, I imagine.
3. Test if the program is currently in test mode via the environment variable and use either `DateTime.now` or e.g. clock
4. Ignore the problem (our actual code will be slightly more streamlined in any case IMO) and accept the fact that coverage will be 0.0x% worse between 0:00 and 0:11.

Either way, I'll maybe have to change every usage of `DateTime.now` for consistency reasons, and I felt like that would have been too big a change not to discuss.